core/templates/dev/head/pages/exploration_editor/editor_tab/StateSolution.js

```diff
@@ -0,0 +1,239 @@
+        !$scope.inlineSolutionEditorIsActive);
+    };
+
+    $scope.$on('stateEditorInitialized', function(evt, stateData) {
+      stateSolutionService.init(
+        editorContextService.getActiveStateName(),
+        stateData.interaction.solution);
+      $scope.interactionHtml = (
+        oppiaExplorationHtmlFormatterService.getInteractionHtml(
+          stateInteractionIdService.savedMemento,
+          explorationStatesService.getInteractionCustomizationArgsMemento(
+            editorContextService.getActiveStateName()),
+          $scope.SOLUTION_EDITOR_FOCUS_LABEL));
+
+      $scope.inlineSolutionEditorIsActive = false;
+      var interactionId = stateInteractionIdService.savedMemento;
+
+      $scope.INTERACTION_SPECS = INTERACTION_SPECS[interactionId];
+
+      if (!StateSolutionHelperService.isSupportedInteraction(interactionId)) {
+        // In this case the interaction UI is constructed by the Solution
+        // object. So interactionHtml is set to null.
+        $scope.interactionHtml = null;
+      }
+      $scope.objectType = StateSolutionHelperService.getInteractionObjectType(
+        interactionId);
+    });
+
+    $scope.getSolutionSummary = function() {
+      var solution = stateSolutionService.displayed;
+      return solution.getSolutionSummary(
+        stateInteractionIdService.savedMemento,
+        responsesService.getAnswerChoices());
+    };
+
+    // This returns false if the current interaction ID is null.
+    $scope.isCurrentInteractionLinear = function() {
+      var interactionId = stateInteractionIdService.savedMemento;
+      return interactionId && INTERACTION_SPECS[interactionId].is_linear;
+    };
+
+    $scope.openAddSolutionModal = function() {
+      alertsService.clearWarnings();
+      $rootScope.$broadcast('externalSave');
+      $scope.inlineSolutionEditorIsActive = false;
+
+      $modal.open({
+        templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+          '/pages/exploration_editor/editor_tab/addSolutionModal.html'),
+        backdrop: 'static',
+        resolve: {
+          objectType: function() {
+            return $scope.objectType;
+          },
+          correctAnswer: function() {
+            return $scope.correctAnswer;
+          },
+          interactionHtml: function() {
+            return $scope.interactionHtml;
+          }
+        },
+        controller: [
+          '$scope', '$modalInstance', 'objectType', 'correctAnswer',
+          'interactionHtml',
+          function($scope, $modalInstance, objectType, correctAnswer,
+            interactionHtml) {
+            $scope.objectType = objectType;
+            $scope.interactionHtml = interactionHtml;
+            $scope.EXPLANATION_FORM_SCHEMA = {
+              type: 'html',
+              ui_config: {}
+            };
+
+            $scope.submitAnswer = function(answer) {
+              $scope.correctAnswer = answer;
+            };
+
+            $scope.tmpSolution = {};
```

Comment:
tmp* format has been using in StateHints and StateResponses (also in the fallbacks file) which is why I chose to follow the same naming style.